cpp/order/Steinhardt.cc

```diff
@@ -30,151 +28,154 @@ void Steinhardt::computeYlm(const float theta,
+            aggregateWl(m_WliOrder, m_Qlmi);
+            m_NormWl = normalizeWl();
+        }
+    }
+    {
         if (m_average)
-            {
-            m_Norm = Steinhardt::normalize();
-            }
+        {
+            m_Norm = normalize();
+        }
         else
-            {
-            m_Norm = Steinhardt::normalize();
-            }
+        {
+            m_Norm = normalize();
         }
     }
+}
 
-void Steinhardt::baseCompute(const box::Box& box,
-							 const locality::NeighborList *nlist,
-							 const vec3<float> *points)
-    {
+void Steinhardt::baseCompute(const box::Box& box, const locality::NeighborList* nlist,
+                             const vec3<float>* points)
+{
     nlist->validate(m_Np, m_Np);
 
-    memset((void*) m_Qlmi.get(), 0, sizeof(complex<float>)*(2*m_l+1)*m_Np);
-    memset((void*) m_Qli.get(), 0, sizeof(float)*m_Np);
-
-    parallel_for(tbb::blocked_range<size_t>(0, m_Np),
-        [=] (const blocked_range<size_t>& r)
-        {
+    parallel_for(tbb::blocked_range<size_t>(0, m_Np), [=](const blocked_range<size_t>& r) {
         const float rminsq = m_rmin * m_rmin;
         const float rmaxsq = m_rmax * m_rmax;
-        const float normalizationfactor = 4*M_PI/(2*m_l+1);
-        const size_t *neighbor_list(nlist->getNeighbors());
-
-		if (!m_average)
-			{
-			bool Qlm_exists;
-			m_Qlm_local.local(Qlm_exists);
-			if (!Qlm_exists)
-				{
-				m_Qlm_local.local() = new complex<float> [2*m_l+1];
-				memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2*m_l+1));
-				}
-			}
+        const float normalizationfactor = 4 * M_PI / (2 * m_l + 1);
+        const size_t* neighbor_list(nlist->getNeighbors());
+
+        if (!m_average)
+        {
+            bool Qlm_exists;
+            m_Qlm_local.local(Qlm_exists);
+            if (!Qlm_exists)
+            {
+                m_Qlm_local.local() = new complex<float>[2 * m_l + 1];
+                memset((void*) m_Qlm_local.local(), 0, sizeof(complex<float>) * (2 * m_l + 1));
+            }
+        }
 
         size_t bond(nlist->find_first_index(r.begin()));
         // for each reference point
         for (size_t i = r.begin(); i != r.end(); i++)
-            {
+        {
             unsigned int neighborcount(0);
             const vec3<float> ref(points[i]);
-            for (; bond < nlist->getNumBonds() && neighbor_list[2*bond] == i; ++bond)
-                {
-                const unsigned int j(neighbor_list[2*bond + 1]);
+            for (; bond < nlist->getNumBonds() && neighbor_list[2 * bond] == i; ++bond)
+            {
+                const unsigned int j(neighbor_list[2 * bond + 1]);
 
                 if (i == j)
-                    {
+                {
                     continue;
-                    }
+                }
 
                 const vec3<float> delta = box.wrap(points[j] - ref);
                 const float rsq = dot(delta, delta);
 
                 if (rsq < rmaxsq && rsq > rminsq)
```

Comment:
See if tests still pass if you remove this `if` statement. We have a problem with this check, see #297. We may need to change the preprocessing we do to the NeighborList to correct the behavior after removing this line.

Reply:
Specifically we need to check the behavior where `rmin > 0`.